src/compiler/checker.ts

```diff
@@ -8168,8 +8166,9 @@ namespace ts {
                     case SyntaxKind.InterfaceDeclaration:
                     case SyntaxKind.TypeAliasDeclaration:
                         const declaration = node as DeclarationWithTypeParameters;
-                        if (declaration.typeParameters) {
-                            for (const d of declaration.typeParameters) {
+                        const typeParameters = declaration.typeParameters || getTypeParametersFromJSDocTemplate(declaration);

```

Comment:
@rbuckton has been using a pattern to use of `getEffectiveType*` to get the TS nodes, or the ones from JSDoc. we can follow the same here and use `getEffectiveTypeParameters`